src/librustdoc/html/render.rs

```diff
@@ -2672,9 +2672,14 @@ fn item_static(w: &mut fmt::Formatter, cx: &Context, it: &clean::Item,
 
 fn item_function(w: &mut fmt::Formatter, cx: &Context, it: &clean::Item,
                  f: &clean::Function) -> fmt::Result {
+    // FIXME: to be removed when "const" keyword is stabilised.
+    let vis_constness = match UnstableFeatures::from_environment().is_nightly_build() {
```

Comment:
Is there a way we could move this check earlier in the process and store the result in the `Cache` or something? That way we wouldn't have to keep poking the environment variables over and over.

The `Context` already has the `codes` field, which can be used as a stand-in for "can i do nightly things", but that won't help the other spot where you put this check, since it doesn't receive a `&Context`.